cmd/erasure-metadata.go

```diff
@@ -476,9 +476,7 @@ func GetInternalReplicationState(m map[string][]byte) ReplicationState {
 
 // getInternalReplicationState fetches internal replication state from the map m
 func getInternalReplicationState(m map[string]string) ReplicationState {
-	d := ReplicationState{
-		ResetStatusesMap: make(map[string]string),
```

Comment:
ResetStatusesMap is used without checking if map is non nil since it is expected to be allocated.Changes needed in bucket-replication-utils.go wherever referenced

Reply:
Isn't line #499 expected to do that @poornas

Reply:
on closer look https://github.com/minio/minio/blob/master/cmd/bucket-replication-utils.go#L383 uses ResetStatusesMap that is allocated, so non-issue here.